node/narwhal/src/helpers/storage.rs

```diff
@@ -534,6 +569,131 @@ impl<N: Network> Storage<N> {
+        // Retrieve the current round in the block.
+        let block_round = block.round().max(1);
+        // If the round in the block is greater than the current round in storage, sync the round.
+        if block_round > self.current_round() {
+            // Update the current round in storage.
+            self.update_current_round(block_round);
+            // Log the updated round.
+            info!("Synced to round {block_round}...");
+        }
+    }
+
+    /// Syncs the batch certificates with the block.
+    fn sync_certificates_with_block(&self, block: Block<N>) {
+        // If the block authority is a subdag, then load the certificates.
+        if let Authority::Quorum(subdag) = block.authority() {
+            // Iterate over the certificates.
+            for certificate in subdag.values().flatten() {
+                // Sync the batch certificate with the block.
+                self.sync_certificate_with_block(&block, certificate);
+            }
+        }
+    }
+
+    /// Syncs the batch certificate with the block.
+    fn sync_certificate_with_block(&self, block: &Block<N>, certificate: &BatchCertificate<N>) {
+        // Skip if the certificate round is below the GC round.
+        if certificate.round() <= self.gc_round() {
+            return;
+        }
+        // If the certificate ID already exists in storage, skip it.
+        if self.contains_certificate(certificate.certificate_id()) {
+            return;
+        }
+        // Retrieve the transmissions for the certificate.
+        let mut missing_transmissions = HashMap::new();
+        // Iterate over the transmission IDs.
+        for transmission_id in certificate.transmission_ids() {
+            // If the transmission ID already exists in the map, skip it.
+            if missing_transmissions.contains_key(transmission_id) {
+                continue;
+            }
+            // If the transmission ID exists in storage, skip it.
+            if self.contains_transmission(*transmission_id) {
+                continue;
+            }
+            // Retrieve the transmission.
+            match transmission_id {
+                TransmissionID::Ratification => (),
+                TransmissionID::Solution(puzzle_commitment) => {
+                    // Retrieve the solution.
+                    match block.get_solution(puzzle_commitment) {
+                        // Insert the solution.
+                        Some(solution) => missing_transmissions.insert(*transmission_id, (*solution).into()),
+                        None => {
+                            error!("Missing solution {puzzle_commitment} in block {}", block.height());
+                            continue;
+                        }
+                    };
+                }
+                TransmissionID::Transaction(transaction_id) => {
+                    // Retrieve the transaction.
+                    match block.get_transaction(transaction_id) {
```

Comment:
Note: This will not be be able to fetch the rejected transactions. 

The transmissionID won't match the transaction id of the `ConfirmedTransaction` in the block.